core/src/test/java/apoc/export/csv/ExportCsvTest.java

```diff
@@ -122,6 +142,29 @@ public void testExportAllCsv() throws Exception {
         assertEquals(EXPECTED, readFile(fileName));
     }
 
+    @Test
+    public void testExportAllCsvWithSample() throws IOException {
+        db.executeTransactionally("CREATE (:User:Sample {`last:Name`:'Galilei'}), (:User:Sample {address:'Universe'}),\n" +
+                "(:User:Sample {foo:'bar'})-[:KNOWS {one: 'two', three: 'four'}]->(:User:Sample {baz:'baa', foo: true})");
+        String fileName = "all.csv";
+        final long totalNodes = 10L;
+        final long totalRels = 3L;
+        final long totalProps = 19L;
+        TestUtil.testCall(db, "CALL apoc.export.csv.all($file, null)", map("file", fileName),
+                (r) -> assertResults(fileName, r, "database", totalNodes, totalRels, totalProps, true));
+        assertEquals(EXP_SAMPLE, readFile(fileName));
+
+        // quotes: 'none' to simplify header testing
+        TestUtil.testCall(db, "CALL apoc.export.csv.all($file, {sampling: true, samplingConfig: {sample: 1}, quotes: 'none'})", map("file", fileName),
```

Comment:
What's the purpose of choosing a sampling of 1 here? Lines below this aren't clear about the purpose.

Reply:
To be sure not to analyze all available properties. 
With low sample (with `1` we are as sure as possible),
the header of the csv is not populated with all possible properties (and this is what I check on line 162 `s.length <17` ). 
However, the values "_id", "_labels", "_start", "_end", "_type" must always be present (line 163)